src/sentry/integrations/slack/sdk_client.py

```diff
@@ -0,0 +1,24 @@
+from slack_sdk import WebClient
+
+from sentry.models.integrations import Integration
+from sentry.services.hybrid_cloud.integration import integration_service
+from sentry.silo.base import SiloMode
+
+
+class SlackSdkClient(WebClient):
```

Comment:
we have a function called track_response_data in all of our in-house ApiClients, i'm wondering how we might do the same for this `SlackClient`... duplicate the method signatures, call super(), and then track the response?

Reply:
Are we sure we need those abstractions or metrics if we aren't already using them?

Reply:
aren't we using them in our integrations health dashboard?

Reply:
`track_response_data` currently tracks all responses. we could also just track errors, if the error rates start climbing above a baseline then we know there's a problem, but it's a little harder to tell without the reference to how many total responses there are

Reply:
Yes, we need total responses to set SLOs. I see it has metrics as well as logger which we really don't need for success responses. Saves us a few buck to only log errors and have metrics only for success.

I also agree with moving as a method here to track response. What's your concern for copying it here?

Reply:
i think to do it i'm going to have to copy the method signature of every method we want to use from the slack_sdk WebClient, call the WebClient function, and track the response after that

like
```
class SlackSdkClient(WebClient):
    def __init__(self, integration_id: int) -> None:
        ....

    def chat_postMessage(...) -> SlackResponse:
        response = super().chat_postMessage(...)
        track_response_data()
        return response
```

and some of these methods have a lot of possible arguments

Reply:
I think you might be looking for https://stackoverflow.com/questions/11349183/how-to-wrap-every-method-of-a-class

Reply:
Got it. Thank you for the explanation. So basically each method needs a wrapper for logging purpose only. You can use args and kwargs for the arguments since we're only passing them.

One ugly alternative is also to have just one wrapper for all of them. Like `def call_and_track_response(method_to_call, args)`

Reply:
updated with the metaclass wrapping